### PR TITLE
Go: Migrate AppenderOrSprinter model to models-as-data

### PR DESCRIPTION
**Pull request by atorralba**
Migrates the existing dataflow model for `fmt.[Append|Sprint][|f|ln]` to models as data to increase its precision regarding access paths.

It also deprecates the old model and creates a new class `AppenderOrSprinterFunc` without the dataflow logic.

### CODE REVIEW

File: go/ql/lib/ext/fmt.model.yml [hunk 1]
```diff
@@ -8,3 +8,14 @@ extensions:
       - ["fmt", "ScanState", True, "Token", "", "", "Argument[-1]", "ReturnValue[0]", "taint", "manual"]
       - ["fmt", "State", True, "Write", "", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
       - ["fmt", "Stringer", True, "String", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
+      - ["fmt", "", True, "Append", "", "", "Argument[0].ArrayElement", "ReturnValue", "taint", "manual"]
```
Comment (owen-mc) [inline go/ql/lib/ext/fmt.model.yml:11]: Even though the signature is `func Append(b []byte, a ...any) []byte`) I think we just want this:
```suggestion
      - ["fmt", "", True, "Append", "", "", "Argument[0]", "ReturnValue", "taint", "manual"]
```
And the same for the other `Append` models

Comment (atorralba) [reply go/ql/lib/ext/fmt.model.yml:11]: I guess we don't expect individual bytes of the `[]byte` parameter to be tainted, right?

**Review by owen-mc — CHANGES_REQUESTED** [review]

**Comment by atorralba** [comment]
All suggestions applied, thanks for the review @owen-mc.

**Comment by atorralba** [comment]
And now the easily-offended tests are also fixed 😄.

**Comment by owen-mc** [comment]
Yes, there are two or three tests that include information about summaries of library functions. There's a PR in progress to fix some of them, so it may be better in future.

**Review by owen-mc — APPROVED** [review]

**Pull request merged**